source/beamline/BioXAS/BioXASZebraPulseControl.cpp

```diff
@@ -167,6 +152,46 @@ bool BioXASZebraPulseControl::outputValue() const
 	return int(outputPulseControl_->value()) == 1;
 }
 
+bool BioXASZebraPulseControl::validTimeValue(double timeValue) const
+{
+	bool result = false;
+
+	if (timeValue >= BIOXASZEBRAPULSECONTROL_PULSE_WIDTH_MIN && timeValue <= BIOXASZEBRAPULSECONTROL_PULSE_WIDTH_MAX)
+		result = true;
+
+	return result;
+}
+
+double BioXASZebraPulseControl::convertTimeValue(double timeValue, double timeUnits, double desiredTimeUnits) const
```

Comment:
Is there a reason why the time units are passed around this class as doubles, rather than the enum or integers?